VideoLocker/src/main/java/org/edx/mobile/view/adapters/CourseOutlineAdapter.java

```diff
@@ -379,52 +354,77 @@ private  View getRowViewForContainer(int position, View convertView, ViewGroup p
+                            }
+                        });
             } else {
-                int inProcessCount = dbStore.getVideosCountBySection(courseId, chapterId, sequentialId, null);
-                int webOnlyCount = dbStore.getWebOnlyVideosCountBySection(courseId, chapterId, sequentialId, null);
-                row.numOfVideoNotDownloaded = totalCount - inProcessCount - webOnlyCount;
-                if (row.numOfVideoNotDownloaded > 0) {
-                    holder.bulkDownload.setVisibility(View.VISIBLE);
-                    holder.numOfVideoAndDownloadArea
-                        .setOnClickListener(new View.OnClickListener() {
+                holder.noOfVideos.setVisibility(View.VISIBLE);
+                setRowStateOnDownload(holder, DownloadEntry.DownloadedState.ONLINE
+                    , new View.OnClickListener() {
                             @Override
                             public void onClick(View downloadView) {
                                 mDownloadListener.download(component.getVideos());
                             }
                         });
-                } else {
-                    holder.bulkDownload.setVisibility(View.GONE);
-                }
             }
         }
 
-        if (AppConstants.offline_flag) {
-            holder.numOfVideoAndDownloadArea.setVisibility(View.GONE);
-            boolean isVideoDownloaded = dbStore.isVideoDownloadedInSection
-                (courseId, chapterId, sequentialId, null);
-            if(isVideoDownloaded)
-            {
-                //TODO - any UI update
-            }else{
-                //TODO - any UI update
-            }
-        } else {
-           //TODO - any UI update?
-        }
-
         return convertView;
     }
 
+    /**
+     * Makes various changes to the row based on a video element's download status
+     *
+     * @param row      ViewHolder of the row view
+     * @param state    current state of video download
+     * @param listener the listener to attach to the video download button
+     */
+    private void setRowStateOnDownload(ViewHolder row, DownloadEntry.DownloadedState state
+            , View.OnClickListener listener) {
+        switch (state) {
+            case DOWNLOADING:
+                row.bulkDownload.setIcon(Iconify.IconValue.fa_spinner);
+                row.bulkDownload.setRotating(true);
+                row.bulkDownload.setIconColorResource(R.color.edx_brand_primary_base);
+                row.numOfVideoAndDownloadArea.setOnClickListener(listener);
+                break;
+            case DOWNLOADED:
+                row.bulkDownload.setIcon(Iconify.IconValue.fa_check);
+                row.bulkDownload.setRotating(false);
+                row.bulkDownload.setIconColorResource(R.color.edx_grayscale_neutral_base);
+                row.numOfVideoAndDownloadArea.setOnClickListener(listener);
+                break;
+            default:
```

Comment:
In my opinion, it's better to declare all cases explicitly in enum switching instead of using defaults, even though it makes sense in this case. Therefore this case could be changed to `ONLINE`.